pkg/operator/ceph/cluster/controller.go

```diff
@@ -271,10 +271,24 @@ func (c *ClusterController) onAdd(obj interface{}) {
 		logger.Warningf("mon count is even (given: %d), should be uneven, continuing", cluster.Spec.Mon.Count)
 	}
 
-	cephVersion, err := cluster.detectCephVersion(cluster.Spec.CephVersion.Image, 15*time.Minute)
-	if err != nil {
-		logger.Errorf("unknown ceph major version. %+v", err)
-		return
+	retryCount := 0
+	retryInterval := time.Duration(versionDetectionRetryInternval) * time.Second
+	// initialize a dummy cephVersion
+	cephVersion := &version.CephVersion{}
+	for {
```

Comment:
if kube-engine takes some time to return the logs, why rerun the entire job (batch job, log collection, version extraction) rather than _wait_ longer on the output from one job?